Sources/Kitura-CredentialsTwitter/CredentialsTwitter.swift

```diff
@@ -0,0 +1,358 @@
+                                onPass: onPass,
+                                inProgress: inProgress)
+        }
+    }
+
+    //MARK: Step 1: Obtaining a request token
+    
+    /// The method gathers the necessary components to create a request to https://api.twitter.com/oauth/request_token. 
+    /// If the request is successful, it will move on to Step 2 which redirects the user. Failure or incorrect payloads
+    /// call `onFailure` or `onPass`, respectively.
+    ///
+    /// - Parameters:
+    ///   - request: The `RouterRequest` object used to get information about the request.
+    ///   - response: The `RouterResponse` object used to respond to the request.
+    ///   - onFailure: The closure to invoke in the case of an authentication failure.
+    ///   - onPass: The closure to invoke when the plugin doesn't recognize the authentication data (usually an
+    ///       authentication token) in the request.
+    ///   - inProgress: The closure to invoke to cause a redirect to the login page in the case of redirecting
+    ///       authentication.
+    func twitterTokenRequest(request: RouterRequest,
+                             response: RouterResponse,
+                             onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                             onPass: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
+                             inProgress: @escaping () -> Void) {
+        var requestOptions: [ClientRequest.Options] = []
+        requestOptions.append(.schema("https://"))
+        requestOptions.append(.hostname("api.twitter.com"))
+        requestOptions.append(.method("POST"))
+        requestOptions.append(.path("oauth/request_token"))
+        var headers = [String : String]()
+        var parameters = [String:String]()
+
+        let nonce = String.nonce
+        let timeStamp = Date().timeIntervalSince1970.roundedString
+
+        parameters["oauth_consumer_key"] = consumerKey
+        parameters["oauth_signature_method"] = "HMAC-SHA1"
+        parameters["oauth_timestamp"] = timeStamp
+        parameters["oauth_nonce"] = nonce
+        parameters["oauth_version"] = "1.0"
+
+        guard let signature = String.oAuthSignature(fromMethod: "POST",
+                                                    urlString: "https://api.twitter.com/oauth/request_token",
+                                                    parameters: parameters,
+                                                    consumerSecret: consumerSecret) else {
+                                                        onFailure(nil, nil)
+                                                        return
+        }
+
+        parameters["oauth_signature"] = signature
+
+        headers["Authorization"] = String.oAuthAuthorizationString(fromParameters: parameters)
+        requestOptions.append(.headers(headers))
+
+        let twitterRequest = HTTP.request(requestOptions) {
+            (optionalResponse) in
```

Comment:
I always put closure parameters in parentheses to keep it consistent whether it's in a tuple or not.